netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -301,7 +345,15 @@ protected void handlerAdded0(ChannelHandlerContext ctx) {
       SSLParameters sslParams = sslEngine.getSSLParameters();
       sslParams.setEndpointIdentificationAlgorithm("HTTPS");
       sslEngine.setSSLParameters(sslParams);
-      ctx.pipeline().addBefore(ctx.name(), /* name= */ null, new SslHandler(sslEngine, false));
+
+      SslHandler sslHandler;
+      if (this.executor != null) {
+        sslHandler = new SslHandler(sslEngine, false, this.executor);
+      } else {
+        sslHandler = new SslHandler(sslEngine, false);
+      }
+      checkNotNull(sslHandler, "sslHandler");
+      ctx.pipeline().addBefore(ctx.name(), null, sslHandler);
```

Comment:
The original code had the param `/* name= */` which is missing here for the 2nd arg. In any case do we need 7 additional lines or can you have the following:
```
ctx.pipeline().addBefore(ctx.name(), /* name= */ null, this.executor != null ?
      new SslHandler(sslEngine, false, this.executor) :
      new SslHandler(sslEngine, false));
```
The `checkNotNull` is gone but why is it required? A `new` call will never return null (the only 2 paths used here).

Reply:
Thank you so much for the suggestion!